association_test.R

```diff
@@ -0,0 +1,62 @@
+#! /usr/bin/env Rscript
+library(argparser)
+library(magrittr)
+argp <- arg_parser("Run association test") %>%
+  add_argument("gds_file", help = "GDS file") %>%
+  add_argument("pheno_file",
+               help = "Phenotype file in annotated dataframe format") %>%
+  add_argument("grm_file", help = "GRM file") %>%
+  add_argument("outcome", help = "Outcome variable name") %>%
+  add_argument("family", help = "Distribution family",
+               default = "gaussian") %>%
+  add_argument("--out_file", help="output file name",
+               default = "assoc.rds") %>%
+  add_argument("--covars",
+               help = "Covariate variable names (space-separated)") %>%
+  add_argument("--variant_id", help = "File with vector of variant IDs") %>%
+  add_argument("--sample_id", help = "File with vector of sample IDs")
+
+argv <- parse_args(argp)
+
+library(SeqArray)
+library(SeqVarTools)
+library(Biobase)
+library(GENESIS)
+
+gds <- seqOpen(argv$gds_file)
+pheno <- readRDS(argv$pheno_file)
+
+if (!is.na(argv$variant_id)) {
+  variant_id <- readRDS(argv$variant_id)
+} else {
+  variant_id <- NULL
+}
+if (!is.na(argv$variant_id)) {
+  sample_id <- readRDS(argv$sample_id)
+} else {
+  sample_id <- NULL
+}
+
+if (!is.na(argv$covars)) {
+  covars <- strsplit(argv$covars, " ") %>% unlist
```

Comment:
The way you pass covariates to the model is with `--covars "Var1 Var2 Var3"` (space-separated string). This seemed like a relatively user-friendly way to do it and it's easy to parse, but I'm not sure it's the best way to go about it

Reply:
The TOPMed pipeline relies heavily on space-separated strings as well. I always felt vaguely uneasy about it but could never think of a better solution.